### PR TITLE
Update what-is-camunda-platform-8.md

### PR DESCRIPTION
**Pull request by felix-mueller**
update whats platform 8, input from @AnnieTalvasto

### CODE REVIEW

**Comment by akeller** [comment]
@christinaausley can you review for style? Looks like a few links are direct or for the C8 staging environment. Also please make sure the voice is informative, not salesy (that was my biggest complaint with the previous iteration of this page).

**Comment by christinaausley** [comment]
I think this is informative while still showcasing the important features of Camunda Platform 8.

I've made a handful of adjustments in my latest commit where things were a touch too wordy or not quite as direct as I'd like to keep things simple for users visiting the page for perhaps the first time. The more concise and direct this page is without sacrificing the core qualities of Camunda Platform 8, the better!

Let me know if you have any questions. Otherwise, LGTM!

**Comment by felix-mueller** [comment]
I thinks linking to other material is always cool

**Comment by christinaausley** [comment]
@felix-mueller Cool 👍 I'll pop that in real quick.

**Review by akeller — APPROVED** [review]
LGTM! Good link adds @christinaausley. And of course, we can always continue iterating on this.
